lib/spack/spack/directives.py

```diff
@@ -317,34 +317,95 @@ def remove_directives(arg):
 
 
 @directive("versions")
-def version(ver, checksum=None, **kwargs):
+def version(
+    ver: str,
+    checksum: Optional[str] = None,  # deprecated years ago
+    *,
+    sha256: Optional[str] = None,
+    sha384: Optional[str] = None,
+    sha512: Optional[str] = None,
+    preferred: Optional[bool] = None,
+    deprecated: Optional[bool] = None,
+    expand: Optional[bool] = None,
+    url: Optional[str] = None,
+    extension: Optional[str] = None,
+    no_cache: Optional[bool] = None,
+    fetch_options: Optional[dict] = None,
+    git: Optional[str] = None,
+    svn: Optional[str] = None,
+    hg: Optional[str] = None,
+    cvs: Optional[str] = None,
+    get_full_repo: Optional[bool] = None,
+    branch: Optional[str] = None,
+    submodules: Optional[bool] = None,
+    submodules_delete: Optional[bool] = None,
+    commit: Optional[str] = None,
+    tag: Optional[str] = None,
+    revision: Optional[str] = None,
+    date: Optional[str] = None,
+    # unsafe checksums
+    md5: Optional[str] = None,
+    sha1: Optional[str] = None,
+    sha224: Optional[str] = None,
```

Comment:
Can these be separated a bit by informative comments?  Grouping by fetch strategy seems like it would be a good thing to do.